b2sum: unstable-2018-06-11 -> 20190724

toonn opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Bump to the most recent tagged release.

I noticed `b2sum` is using a seemingly random revision, figured it's better to stick to the latest tagged release since it's newer.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

toonn committed:
b2sum: unstable-2018-06-11 -> 20190724

Bump to the most recent tagged release.

AndersonTorres commented on pkgs/tools/security/b2sum/default.nix:
```diff
@@ -2,15 +2,15 @@
 with lib;
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

AndersonTorres commented on pkgs/tools/security/b2sum/default.nix:
```diff
@@ -2,15 +2,15 @@
     repo = "BLAKE2";
-    rev = "320c325437539ae91091ce62efec1913cd8093c2";
-    sha256 = "E60M9oP/Sdfg/L3ZxUcDtUXhFz9oP72IybdtVUJh9Sk=";
+    rev = "${version}";
```
```suggestion
    rev = finalAttrs.version;
```

AndersonTorres reviewed (changes requested):
Use rec-less style as shown, and edit the last line to `})`.

toonn committed:
b2sum: Prefer finalAttrs over rec

This came up in review of PR #198456 and should make overriding more consistent.

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

SuperSandro2000 commented:
@AndersonTorres there is no real advantage of using finalAttrs over rec. Overlays completely overwrite src most of the time anyway. But it is still pretty new, most likely has some weird edge cases and performance bad when nested. I would highly prefer if we don't use it villy nilly in some packages.

toonn commented:
Should I revert the change to using `finalAttrs`?

AndersonTorres commented:
> @AndersonTorres there is no real advantage of using finalAttrs over rec.

Well, getting rid of rec is an advantage.

> Overlays completely overwrite src most of the time anyway.

You are merely stating that it is harmless...

> But it is still pretty new, most likely has some weird edge cases and performance bad when nested.

This is a good motivation to battle-test it, not to avoid it.

> I would highly prefer if we don't use it villy nilly in some packages.

I am for using it and report such edge cases. Burying an improvement on the codebase is not a good thing.

> 
> Also from a reviewer perspective it is unnecessary to suggest it here.

You have a point.

AndersonTorres reviewed (changes requested):
Change the last line to `})` instead of mere `}`.

SuperSandro2000 commented:
> Well, getting rid of rec is an advantage.

Using rec is perfectly fine, not like it bites. Recursively changing everything is not always desirable especially if people use pname everywhere and then you change pname to add a ``-headless``.
